### PR TITLE
Fixes for the PETSc 3.17 series

### PR DESCRIPTION
**Pull request by cticenhour**
## Reason
Updating to newer versions of PETSc enables easier maintenance and new features. But MOOSE requires some fixes to enable updating PETSc. 

## Design
Fixup MOOSE build and operation deficiencies for PETSc 3.17 series.

## Impact
Easier upgrade path for PETSc.

### CODE REVIEW

**Comment by loganharbour** [comment]
Throwing the container recipe on here just to see if it works.

**Comment by lindsayad** [comment]
We want to do 3.17 as opposed to 3.18? I don't know exactly when the automated `-options_left` parameter got applied in PETSc but it's super nice

**Comment by cticenhour** [comment]
The choice of 3.17 was mostly precedent based. We've lagged one minor revision behind for a while. If there's enough interest in jumping up to 3.18, it's pretty easy to reconfigure this PR.

**Comment by lindsayad** [comment]
Eh, 3.17 is fine

**Comment by cticenhour** [comment]
@oanaoana do you have an opinion here?

**Comment by cticenhour** [comment]
Changing the scope of this PR to not include the submodule update for 3.17. It's been long enough since I tried to take this on that jumping to 3.18 seems a better option. I will merge the code fixes for the 3.17 and later changes and move on to a new PR with 3.18.

File: framework/src/utils/PetscSupport.C [hunk 1]
```diff
@@ -676,7 +682,7 @@ storePetscOptions(FEProblemBase & fe_problem, const InputParameters & params)
 #endif
 
       if (option.first == "-matptap_via" || option.first == "-matmaijptap_via" ||
-          option.first == "-matproduct_ptap_via")
+          option.first == "-matproduct_ptap_via" || option.first == "-matproduct_ptap_via")
```
Comment (lindsayad) [inline framework/src/utils/PetscSupport.C:685]: Are my eyes deceiving me? These look like the same string?

Comment (cticenhour) [reply framework/src/utils/PetscSupport.C:685]: They definitely are - should be `-mat_product_algorithm` instead. Thanks for the catch!

**Review by lindsayad — APPROVED** [review]

**Pull request merged**